Update typescript to 2.8.0

himdel opened the pull request:
In preparation for the webpack5 upgrade (https://github.com/ManageIQ/manageiq-ui-classic/pull/3776/), we need to upgrade typescript first.

~~ts-loader 3.5.0 is not the newest version, it's the newest version which works with webpack3 (https://github.com/TypeStrong/ts-loader/issues/729)~~

~~Unfortunately updating to new typescript breaks any files which include a .tsx file which includes react. (Typescript can't seem to handle a conditional dynamic export in React sources.)
Fortunataly this is just one file in the whole project - converting back to .jsx. (Will try undoing when we are on webpack 5.)~~

Updated to use awesome-typescript-loader (4.0.0; 5.0.0 doesn't work with current webpack).

himdel commented:
(The error we get with that `.tsx` file as it is:

```
ERROR in ./app/javascript/miq-component/react-blueprint.tsx
[tsl] ERROR in /home/himdel/manageiq-ui-classic/app/javascript/miq-component/react-blueprint.tsx(1,8)
      TS1192: Module '"/home/himdel/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/bundler/gems/manageiq-graphql-d9ea9b128e2f/node_modules/react/index"' has no default export.

ERROR in ./app/javascript/miq-component/react-blueprint.tsx
[tsl] ERROR in /home/himdel/manageiq-ui-classic/app/javascript/miq-component/react-blueprint.tsx(2,8)
      TS1192: Module '"/home/himdel/.rbenv/versions/2.4.3/lib/ruby/gems/2.4.0/bundler/gems/miq_v2v_ui_plugin-1575ee6e3627/node_modules/react-dom/index"' has no default export.
```
)

karelhala commented:
We should ditch `ts-loader` and use `awesome-typescript-loader`, the first one seems to loose the traction and is not good. On the other hand we have to test if the second one works correctly with webpack 4 (looks like v 5+ of awesome ts loader is gaprindashvili). For our settings use `"awesome-typescript-loader": "4.0.0"`

himdel commented:
@miq-bot add_label wip

himdel commented:
@miq-bot remove_label wip

Updated to use `awesome-typescript-loader` (4.0.0; 5.0.0 doesn't work with current webpack).

And removed the `tsx` to `jsx` rewrite since it's no longer needed.
